couler/steps/tensorflow.py

```diff
@@ -0,0 +1,186 @@
+import copy
+import uuid
+
+import pyaml
+
+import couler.argo as couler
+from couler.core import utils
+
+container_template = {"name": "tensorflow", "image": "", "command": ""}
+
+pod_template = {
+    "replicas": 1,
+    "restartPolicy": "",
+    "template": {"spec": {"containers": []}},
+}
+
+manifest_template = {
+    "apiVersion": '"kubeflow.org/v1alpha2"',
+    "kind": '"TFJob"',
+    "metadata": {"name": ""},
+    "spec": {"cleanPodPolicy": "", "tfReplicaSpecs": {}},
+}
+
+pod_types = {"Chief", "PS", "Worker"}
+
+
+def _validate_pod_params(
+    pod_type=None, image=None, replicas=0, restart_policy=None
+):
+
+    if pod_type not in pod_types:
+        raise ValueError("Invalid value %s for parameter pod_type." % pod_type)
+    if replicas == 0:
+        raise ValueError("Parameter replicas value should be more than 0.")
+    if image is None:
+        raise ValueError("Parameter image should not be None.")
+    if pod_type == "Chief" and replicas > 1:
+        raise ValueError("Chief pod's replicas should be 1.")
+    if restart_policy is None:
+        raise ValueError("Parameter restart_policy should not be None.")
+
+
+def _generate_pod(
```

Comment:
Move some of these methods to utility since they will be reused by other Kubeflow jobs?

Reply:
good point, let me check this and move this into core part.

Reply:
We can move this when we support additional Kubeflow jobs in follow-up PRs.